### PR TITLE
possibility to divide acceptance tests suites into parts on the fly

### PR DESCRIPTION
**Pull request by individual-it**
## Description
make it possible to divide a test run of multiple suites on the fly into parts

## Related Issue
https://github.com/owncloud/encryption/pull/34

## Motivation and Context
We need to be able to run all suites, but running them all does take too long, so we want to divide and parallelize the test runs. In core we simply run one suite in a drone job. But when doing that for apps, we need to be able to divide the test run without knowing how many and which suites are available.

## How Has This Been Tested?
- run only one test `./run.sh features/apiMain/tags.feature:18`
- run all tests of a feature file `./run.sh features/apiMain/appmanagement.feature`
- run one full suite `./run.sh --suite apiSharingNotifications`
- run all tests of the same type `./run.sh --type webui`
- run first part of a small subset of suites `./run.sh --type webui --part 1 15`
- run last part of a small subset of suites `./run.sh --type webui --part 15 15`
- run tagged tests in a subset of suites `./run.sh --tags @myNewTag --part 1 10`
- run tagged tests in all suites of one type `./run.sh --tags @myNewTag `
- divide into more parts than there are suites, run first part of the subset `./run.sh --part 1 12`
- divide into more parts than there are suites, run the first non-existing part `./run.sh --part 11 12`
- try to run a part-no that is higher than the count of divisions `./run.sh --part 11 10`
- divide into equal parts, run first part `./run.sh --part 1 5`
- divide into equal parts, run second part `./run.sh --part 2 5`
- divide into equal parts, run last part `./run.sh --part 5 5`
- divide into unequal parts, run first part `./run.sh --part 1 3`
- divide into unequal parts, run last part `./run.sh --part 3 3`
- divide into parts, make a test in first suite fail `./run.sh --part 1 5`
- divide into parts, make a test in last suite fail `./run.sh --part 1 5`
- set settings by env. variables `RUN_PART=1 DIVIDE_INTO_NUM_PARTS=5 ./run.sh`
- check drone details of core

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/run.sh [hunk 1]
```diff
@@ -23,6 +23,9 @@ BEHAT=${OC_PATH}lib/composer/bin/behat
 #            testing app. We have to assume it is already enabled.
 # --show-oc-logs - tail the ownCloud log after the test run
 # --norerun - do not rerun failed webUI scenarios
+# --part - run a subset of scenarios, need two numbers.
+#          first number: which part to run
+#          second number: in how many party to devide the set of scenarios
```
Comment (phil-davis) [inline tests/acceptance/run.sh]: in how many parts to divide the set of scenarios

File: tests/acceptance/run.sh [hunk 2]
```diff
@@ -59,6 +62,16 @@ do
 			BROWSER="$2"
 			shift
 			;;
+		--part)
+			RUN_PART="$2"
+			DEVIDE_INTO_NUM_PARTS="$3"
```
Comment (phil-davis) [inline tests/acceptance/run.sh]: s/DEVIDE/DIVIDE

File: tests/acceptance/run.sh [hunk 3]
```diff
@@ -23,6 +23,9 @@ BEHAT=${OC_PATH}lib/composer/bin/behat
 #            testing app. We have to assume it is already enabled.
 # --show-oc-logs - tail the ownCloud log after the test run
 # --norerun - do not rerun failed webUI scenarios
+# --part - run a subset of scenarios, need two numbers.
+#          first number: which part to run
+#          second number: in how many parts to devide the set of scenarios
```
Comment (phil-davis) [inline tests/acceptance/run.sh]: s/devide/divide

File: tests/acceptance/run.sh [hunk 4]
```diff
@@ -361,11 +465,31 @@ then
...
+	then
+		ALL_SUITES=`find features/ -type d -iname ${ACCEPTANCE_TEST_TYPE}* | sort | cut -d"/" -f2`
+		COUNT_ALL_SUITES=`echo "${ALL_SUITES}" | wc -l`
+		if [ ${RUN_PART} -gt ${COUNT_ALL_SUITES} ]
+		then
+			echo "there are only ${COUNT_ALL_SUITES} suites, nothing to do"
+			teardown
+			exit 0
+		fi
+		#devide the suites and round up
```
Comment (phil-davis) [inline tests/acceptance/run.sh]: s/devide/divide

**Review by phil-davis — COMMENTED** [review]
Just 2 places in comments where "divide" has a typo.

**Comment by individual-it** [comment]
@phil-davis fixed the typos

**Review by phil-davis — APPROVED** [review]

**Comment by individual-it** [comment]
backport in #32271

**Pull request merged**